apps/fortuna/src/state.rs

```diff
@@ -59,11 +60,14 @@ impl PebbleHashChain {
         input.extend_from_slice(random);
 
         let secret: [u8; 32] = Keccak256::digest(input).into();
-        Ok(Self::new(
-            secret,
-            chain_length.try_into()?,
-            sample_interval.try_into()?,
-        ))
+
+        let chain_length: usize = chain_length.try_into()?;
+        let sample_interval: usize = sample_interval.try_into()?;
+        let hash_chain = spawn_blocking(move || Self::new(secret, chain_length, sample_interval))
```

Comment:
I don't understand how this helps. This ran synchronously before, so how does the blocking thread help exactly? (I don't totally understand blocking threads)